src/test/java/org/cactoos/scalar/ScalarOfTest.java

```diff
@@ -51,29 +53,68 @@ void worksWithCallable() {
     @Test
     void worksWithRunnable() {
         final Object obj = new Object();
+        final Object result = new Object();
+        final AtomicReference<Object> done = new AtomicReference<>();
         new Assertion<>(
-            "must hold the same value as given",
-            new ScalarOf<>(new RunnableOf(ignored -> { }, "ignored"), obj),
-            new ScalarHasValue<>(obj)
+            "Must convert Runnable into Scalar",
+            new ScalarOf<>(
+                new RunnableOf(
+                    () -> {
+                        done.set(result);
+                    }
+                ),
+                obj
+            ),
+            new MatcherOf<>(
+                scalar -> {
+                    final Object res = scalar.value();
+                    return res.equals(obj) && done.get().equals(result);
+                }
+            )
         ).affirm();
     }
 
     @Test
     void worksWithFunc() {
-        final Object obj = new Object();
+        final Object ipt = new Object();
         new Assertion<>(
-            "must hold the same value as given by func",
-            new ScalarOf<>(new FuncOf<>(new Constant<>(obj)), "ignored"),
-            new ScalarHasValue<>(obj)
+            "Must convert Func into Scalar",
+            new ScalarOf<>(new FuncOf<>(input -> input), ipt),
+            new ScalarHasValue<>(ipt)
         ).affirm();
     }
 
     @Test
     void worksWithProc() {
+        final Object ipt = new Object();
+        final Object result = new Object();
+        final AtomicReference<Object> done = new AtomicReference<>();
+        new Assertion<>(
+            "Must convert Func into Scalar",
+            new ScalarOf<>(
+                new ProcOf<>(
+                    input -> {
+                        done.set(input);
+                    }
+                ),
+                ipt,
+                result
+            ),
+            new MatcherOf<>(
+                scalar -> {
+                    final Object res = scalar.value();
+                    return res.equals(result) && done.get().equals(ipt);
```

Comment:
@victornoel I think that in this context, it is better to use `==` because we actually get objects `result` and `ipt`.